File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.js
function solution(arr) {
    //1. 가장 작은 수 확인 => 2. 배열에서 가장 작은수 인덱스 확인 => 3. 인덱스 제거 => 빈 배열일때 -1 추가  
    let min = Math.min(...arr); 
    
    let answer = arr.filter(num => {
       return num !== min 
    });
    if(answer.length === 0){
        answer.push(-1);
        return answer ; 
    }
    
    return answer ; 
    
    
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
import java.util.* ; 
class Solution {
    public int[] solution(int[] arr) { 
        List<Integer> list = new ArrayList<>() ; 
        int num = arr[0] ;
       
         if(arr.length == 1) {
            return new int[]{-1} ;
        }
          for(int i : arr){
            if(num > i) {
                num = i ; 
            }
        }
      
      
        for(int k = 0 ; k<arr.length; k++){
            if(num != arr[k]){        
               list.add(arr[k]) ; 
            }
            
        }
        int[] result = new int[list.size()]; 
        for(int i = 0 ; i<list.size()  ; i++){
            result[i] = list.get(i) ; 
        }

        return result ; 
      
    }
}